iothub/device/src/Transport/Amqp/AmqpUnit.cs

```diff
@@ -137,10 +134,11 @@ internal async Task<AmqpIotSession> EnsureSessionIsOpenAsync(CancellationToken c
             }
             catch (OperationCanceledException)
             {
-                throw new IotHubClientException(
-                    "Failed to enter the semaphore required for opening an AMQP session.", 
-                    true, 
-                    IotHubStatusCode.Timeout);
+                if (Logging.IsEnabled)
+                {
+                    Logging.Error(this, "Failed to enter the semaphore required for opening an AMQP session.", nameof(EnsureSessionIsOpenAsync));
+                }
```

Comment:
We choose to forgo braces for logging statements to keep them from blowing up file sizes (lines).

```suggestion
                if (Logging.IsEnabled)
                    Logging.Error(this, "Failed to enter the semaphore required for opening an AMQP session.", nameof(EnsureSessionIsOpenAsync));
```